ContentBrowser/src/main/java/com/amazon/android/contentbrowser/ContentBrowser.java

```diff
@@ -1653,6 +1684,13 @@ else if (subscriptionRequired && purchaseRequired) {
                         R.string.watch_now_1, R.string.watch_now_2));
             }
         }
+
+        if(registrationRequired) {
+            contentActionList.add(createActionButton(CONTENT_REGISTRATION_REQUIRED,
+                R.string.action_signup_to_watch1, R.string.action_signup_to_watch2));
+            return contentActionList;
```

Comment:
i think its fine, since we need to show only a single button here if the user is not authenticated, once the user authenticates, we update the action buttons